### ISSUE TITLE
Close() with MessageBoxResult isn't implemented

### ISSUE THREAD

**Issue by Difegue** [lepoco/wpfui#1046]
### Describe the bug

Overriding `OnButtonClick()` in the WPFUI MessageBox doesn't allow me to just call `Close()`, throwing an InvalidOperationException asking me to use the version with MessageBoxResult.  

Which would be fine... if that method existed? `MessageBox` doesn't seem to have a Close(MessageBoxResult) method implemented, instead relying on the WPF `Close()` internally. 

### To Reproduce

Try subclassing `MessageBox` and overriding `OnButtonClick()`

### Expected behavior

Being able to Close() the popup myself without having to downcast to `Window`

### OS version

Irrelevant

### .NET version

Irrelevant

### WPF-UI NuGet version

3.0.3

### Additional context

https://github.com/lepoco/wpfui/blob/bf3c4a6be4ed05a3ab3f34fd0e88b2fe191714bb/src/Wpf.Ui/Controls/MessageBox/MessageBox.cs#L302

**Comment by Levvy055** [comment]
Workaround(or intended) is to set `DialogResult = true` to close window

**Comment by Veritera** [comment]
@Difegue If this issue is fixed and the current version, please close this issue.

**Comment by Difegue** [comment]
nope, still a thing: https://github.com/lepoco/wpfui/blob/main/src%2FWpf.Ui%2FControls%2FMessageBox%2FMessageBox.cs#L297
